pkg/cli/formatter/table.go

```diff
@@ -50,242 +76,226 @@ func (f *TableFormatter) Format(resp *http.Response) ([]byte, error) {
 		return nil, nil
 	}
 
-	var data any
+	var (
+		r         = gjson.ParseBytes(body)
+		formatted string
+	)
 
-	err = json.Unmarshal(body, &data)
-	if err != nil {
-		return nil, err
+	switch {
+	case r.IsObject():
+		result := gjson.GetBytes(body, "items")
+		if result.Exists() {
+			formatted = f.resourceItems([]byte(result.Raw))
+		} else {
+			formatted = f.resourceItem(body)
+		}
+
+	case r.IsArray():
+		formatted = f.resourceItems(body)
 	}
 
-	switch reflect.TypeOf(data).Kind() {
-	case reflect.Map:
-		m, ok := data.(map[string]any)
-		if !ok {
-			return nil, errors.New("can't decode response in table, use json or yaml format instead")
-		}
+	return []byte(formatted), nil
+}
 
-		if len(m) == 0 {
-			return []byte{}, nil
-		}
+func (f *TableFormatter) resourceItems(body []byte) string {
```

Comment:
It seems that both `resourceItems` and `resourceItem` have similar logic to get table and it's string, shall them exact to a common function?

Reply:
Abstracted the common parts into function renderColumns with input columns and data.